rbd/admin/admin.go

```diff
@@ -0,0 +1,59 @@
+// +build !nautilus
+
+package admin
+
+import (
+	"fmt"
+
+	"github.com/ceph/go-ceph/internal/commands"
+)
+
+// RadosCommander provides an interface to execute JSON-formatted commands that
+// allow the cephfs administrative functions to interact with the Ceph cluster.
+type RadosCommander = commands.RadosCommander
+
+// RBDAdmin is used to administrate rbd volumes and pools.
+type RBDAdmin struct {
+	conn RadosCommander
+}
+
+// NewFromConn creates an new management object from a preexisting
+// rados connection. The existing connection can be rados.Conn or any
+// type implementing the RadosCommander interface.
+func NewFromConn(conn RadosCommander) *RBDAdmin {
+	return &RBDAdmin{conn}
+}
+
+// LevelSpecifier provides an interface to convert any supporting type
+// to a ceph "levelspec".
```

Comment:
An interface to a string method is too weak I would say, because it basically enforces no structure at all. You could feed everything to the consuming functions, that has this function implemented, without having any control over the structure of the string that is returned. So such an interface doesn't gain much. I suggest that the `LevelSpec()` returns a `LevelSpec` (what an coincidence ;-) ) which implements Stringer. Then we have control over what data actually must be returned and can format it correctly in the `String()` method.

Reply:
IIRC, the reason its is a string is that I don't actually have any control over what kind of stuff ceph accepts. On the "wire" the levelspec is a string (a string json argument). I wanted to build in an "escape hatch" in case go-ceph's way of constructing the levelspec is too limited.  I thought the interface was an OK compromise as most users would use the type we supply, but if its not sufficient they could pass any arbitrary string thru to ceph. If they go so far as to define their own type that passes broken stuff thru that's their own problem. :-)
Does that make sense?